src/binaryen-c.h

```diff
@@ -2051,22 +2051,15 @@ BINARYEN_API void BinaryenRemoveEvent(BinaryenModuleRef module,
 
 // Function table. One per module
 
-// TODO: Add support for multiple segments in BinaryenSetFunctionTable.
-BINARYEN_API void BinaryenSetFunctionTable(BinaryenModuleRef module,
-                                           BinaryenIndex initial,
-                                           BinaryenIndex maximum,
-                                           const char** funcNames,
-                                           BinaryenIndex numFuncNames,
-                                           BinaryenExpressionRef offset);
-BINARYEN_API int BinaryenIsFunctionTableImported(BinaryenModuleRef module);
+BINARYEN_API int BinaryenIsTableImported(BinaryenModuleRef module);
```

Comment:
Wondering whether APIs assuming that there is a single function table in a module are still needed, once there can be multiple tables. My expectation would be that one can add multiple tables, and table operations, like checking or setting whether a table is imported, adding segments, walking segments etc. would then operate upon a `BinaryenTableRef`. Or, to obtain all segments in a module, one would walk over all tables, and then over each segment in each table.

Reply:
You're right about this API, it should take a `BinaryenTableRef` for sure. Just note that since we also have passive elem segments that do not belong to a table, obtaining all segments by walking over all tables is not possible.

Reply:
I see, hmm. That would actually make in favor of representing segments the binary format way, as a list of segments belonging to a table or not, with some passive, plus perhaps a convenience API to walk all segments attached to a specific table, if that is possible. Hmmhmhm.

Reply:
That's right, that's how Binaryen will represent elem segments internally with this PR. I'm just not sure what can be done for iterating over the segments that belong to a specific table. The problem is that tables don't have references to their segments in the IR, so we have to iterate over all elem segments in the module. One thought was to return all elem refs for that table in an array, but I don't think any other API behaves like that.

Reply:
Agree, I guess the convenience API isn't strictly necessary in this PR, as long as iterating segments is generally possible. Can still look into that later, once we have a better idea what API users would actually need.